File: app/src/main/java/com/vkochenkov/snakegame/activities/MainActivity.java
package com.vkochenkov.snakegame.activities;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.vkochenkov.snakegame.R;

public class MainActivity extends AppCompatActivity {

    private Button btnPlay;
    private Button btnExit;
    private Button btnShare;
//    private Button btnOptions;
//    private Button btnInfo;


    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnPlay = findViewById(R.id.btn_play);
        btnExit = findViewById(R.id.btn_exit);
        btnShare = findViewById(R.id.btn_share);

        btnPlay.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(MainActivity.this, GameScreenActivity.class);
                startActivity(intent);
            }
        });
        btnShare.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(Intent.ACTION_SEND);
                intent.setType("text/plain");
                intent.putExtra(Intent.EXTRA_TEXT, getString(R.string.share_link_text));
                startActivity(Intent.createChooser(intent, null));
            }
        });
        btnExit.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                finish();
            }
        });
    }
}

File: app/src/main/java/com/vkochenkov/snakegame/activities/MainActivity.kt
package com.vkochenkov.snakegame.activities

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import com.vkochenkov.snakegame.R

class MainActivity : AppCompatActivity() {

    private lateinit var btnPlay: Button
    private lateinit var btnExit: Button
    private lateinit var btnShare: Button
    private lateinit var btnSettings: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initFields()
        setOnClickListeners()
    }

    private fun setOnClickListeners() {
        btnPlay.setOnClickListener {
            val intent = Intent(this@MainActivity, SnakeGameScreenActivity::class.java)
            startActivity(intent)
        }
        btnSettings.setOnClickListener {
            val intent = Intent(this@MainActivity, SettingsActivity::class.java)
            startActivity(intent)
        }
        btnShare.setOnClickListener{
            val intent = Intent(Intent.ACTION_SEND)
            intent.type = "text/plain"
            intent.putExtra(Intent.EXTRA_TEXT, getString(R.string.share_link_text))
            startActivity(Intent.createChooser(intent, null))
        }
        btnExit.setOnClickListener{ finish() }
    }

    private fun initFields() {
        btnPlay = findViewById(R.id.btn_play)
        btnExit = findViewById(R.id.btn_exit)
        btnShare = findViewById(R.id.btn_share)
        btnSettings = findViewById(R.id.btn_settings)
    }
}